recipes-core/date-sync/init

```diff
@@ -1,42 +1,81 @@
 #!/bin/sh
-#
 ### BEGIN INIT INFO
-# Provides:		date-sync
-# Required-Start:	$remote_fs $syslog $networking
-# Required-Stop:	$remote_fs $syslog
-# Default-Start:	2 3 4 5
-# Default-Stop:		1
-# Short-Description:	Start and stop the date-sync daemon
+# Provides:             date-sync
+# Required-Start:       $remote_fs $syslog $networking
+# Required-Stop:        $remote_fs $syslog
+# Default-Start:        2 3 4 5
+# Default-Stop:         1
+# Short-Description:    Start and stop the date-sync daemon
 ### END INIT INFO
-#
 
 PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
 NAME=date-sync
 DESC="Date Sync"
 LOGFILE=/tmp/date-sync.log
+PIDFILE=/var/run/date-sync.pid
 
-start() {
-	echo -n "Starting $DESC: "
+# Configuration
+NTP_SERVER="pool.ntp.org"
+SYNC_INTERVAL=60  # Run every 60 seconds (1 minute)
+MAX_LOG_SIZE=524288  # 512KB (0.5MB) in bytes
 
-	# update time by fetching the latest one from ntp
-	ntpd -q -p pool.ntp.org
+log() {
+    echo "$(date +"%Y-%m-%d %H:%M:%S"): $1" >> $LOGFILE
+    if [ -f "$LOGFILE" ] && [ $(stat -c %s "$LOGFILE") -gt $MAX_LOG_SIZE ]; then
+        tail -n 1000 "$LOGFILE" > "${LOGFILE}.tmp" && mv "${LOGFILE}.tmp" "$LOGFILE"
+    fi
+}
 
-	# get current time
-	current_time=$(date)
+sync_time() {
+    ntpd_output=$(ntpd -n -q -d -p $NTP_SERVER 2>&1)
+    if [ $? -eq 0 ]; then
+        offset=$(echo "$ntpd_output" | grep "offset:" | tail -n 1 | sed 's/.*offset:\([^ ]*\).*/\1/')
+        if [ -n "$offset" ]; then
+            log "Time synchronized with $NTP_SERVER successfully. Offset: $offset seconds"
```

Comment:
Why not just pipe the output as is only filtering for the relevant lines? There's more info than just offset (which arguably is the most relevant one)

Reply:
I can remove the success and failure. But this above won't work because the ntpd -n -q -d -p causes it to trigger the output on the terminal of the logged in user.
Also it gives several lines output with different data and offset is part of the logs that are outputted.
This way I did above, it suppresses the output to the variable instead and I filter the offset out of it